In [1]:
from math import factorial

# Задача 1

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, 
если в этот день должно быть 4 разных занятия?

Решение :
в пн 4 разных занятия из 10 занятий, значит используем формулу для размещений,тут важен порядок

In [2]:
def arr(n, k):
    return (factorial(n) / factorial(n - k))

In [3]:
arr(10, 4)

5040.0

Ответ: 5040 способов

# Задача 2

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Решение: всего в колоде 4 туза, значит вероятность вытянуть хотя бы 1 туза - 4 к 52

In [4]:
ver_tuz=(4/52)
ver_tuz

0.07692307692307693

найдм сколько всего комбинации и умножим на вероятность достать туза

порядок не важен, используем формулу для сочетаний

In [11]:
def combinations(n, k):
    return (factorial(n) / (factorial(k) * factorial(n - k)))

считаем отдельно число вариантов извлечь туза*на число оставшихся карт в колоде и складываем их

In [23]:
k=combinations(48,3)*combinations(4,1)+combinations(48,2)*combinations(4,2)+combinations(48,1)*combinations(4,3)+combinations(48,0)*combinations(4,4)
k

76145.0

новый ответ 76145

# Задача 4

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. 
Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

общее кол-во способов извлечь 3 билета из 60

In [29]:
combinations(60, 3)

34220.0

кол-во способов извлечь 3 билета из 50

In [30]:
combinations(50, 3)

19600.0

In [31]:
a=combinations(50, 3)/combinations(60, 3)
a

0.5727644652250146

извлекаем 2 из 50 и 1 из 10

In [38]:
b=combinations(50, 2)*combinations(10, 1)
b

12250.0

In [48]:
b/combinations(60, 3)

0.35797779076563413

# Задача 5

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх.
Являются ли эти события независимыми?


если P(A)=P(A|B) - события независимы, по условию

P(A)=четные 2,4,6 = вер 3 к 6 = 0.5
P(B)= появление числа больше 3, т.е. 4,5,6 = вер = 3 к 6= 0.5
p(A|B)=появление четного числа при условии выпадения на кубике больше 3,те. 4,5,6=2/3

In [28]:
c=(0.5*0.5) / 0.5
c

0.5

Ответ, тк вероятности событий не равны P(A|B)=2/3, а P(A)=0.5, события зависимы.

# Задача 6 (Дополнительно)

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1 % населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99 % больных этой болезнью и всего лишь в 1 % случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

P(A|B) = P(A*B) / P(B) = P(A)*P(B|A) / P(B) = P(A)*P(B|A) / P(a1)⋅P(b|a1)+P(a2)⋅P(b|a2)
P(a|b)-болеть, при полож.тесте
P(b)+ тест верный - 99%
P(a)-болеть
P(B)

P(A)=P(B1)⋅P(A|B1)+P(B2)⋅P(A|B2)

P(b)=P(a1)⋅P(b|a1)+P(a2)⋅P(b|a2)